src/Microsoft.Azure.WebJobs.Host/JobHost.cs

```diff
@@ -24,7 +25,7 @@ namespace Microsoft.Azure.WebJobs
     /// A <see cref="JobHost"/> is the execution container for jobs. Once started, the
     /// <see cref="JobHost"/> will manage and run job functions when they are triggered.
     /// </summary>
-    public class JobHost : IJobHost, IDisposable, IJobInvoker
+    public class JobHost : IJobHost, IHostedService, IDisposable, IJobInvoker
```

Comment:
It feels odd to have this implement `IHostedService` when we don't actually register as an `IHostedService` anywhere... and in fact we host it inside of another `IHostedService`. 

@fabiocav -- why did we do it this way?

Reply:
Yeah, I didn't see any major reason to have JobHostService separate from JobHost, other than perhaps the missing CancellationToken on IJobHost.StopAsync.

JobHostService is a public class, or I'd suggest just removing it entirely. Another option would be to leave the class but change WebJobsHostBuilderExtensions to register JobHost directly rather than JobHostService as the IHostedService (also moving the ILogger logic that's the only thing currently left in JobHostService).

Those both seem like slightly larger changes, so my inclination would be to do what's in the current PR, but I'm open to options here.

Reply:
Per offline discussion, I've split that commit into its own PR (#2372) so we can merge this part in the meantime.